File: from-1701-to-1800/1769-minimum-number-of-operations-to-move-all-balls-to-each-box/solution.py
class Solution:
    def minOperations(self, boxes: str) -> List[int]:
        n = len(boxes)
        result = [0] * n

        left_moves = left_balls = 0
        for i in range(n):
            result[i] = left_moves
            if boxes[i] == '1':
                left_balls += 1
            left_moves += left_balls

        right_moves = right_balls = 0
        for i in range(n -1, -1, -1):
            result[i] += right_moves
            if boxes[i] == '1':
                right_balls += 1
            right_moves += right_balls

        return result

        return result

File: from-1701-to-1800/1769-minimum-number-of-operations-to-move-all-balls-to-each-box/solution.go
func minOperations(boxes string) []int {
    n := len(boxes)
    result := make([]int, n)

    leftBalls, leftMoves := 0, 0
    for i := 0; i < n; i ++ {
        result[i] += leftMoves
        if boxes[i] == '1' {
            leftBalls += 1
        }
        leftMoves += leftBalls
    }

    rightBalls, rightMoves := 0, 0
    for i := n - 1; i >= 0; i -- {
        result[i] += rightMoves
        if boxes[i] == '1' {
            rightBalls += 1
        }
        rightMoves += rightBalls
    }

    return result
}